save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt

```diff
@@ -0,0 +1,258 @@
+                                            postImageUpload(event.target)
+                                        }
+                                    }
+                                    ariaLabel = "Change user's avatar"
+                                    img {
+                                        className =
+                                                ClassName("avatar avatar-user width-full border color-bg-default rounded-circle")
+                                        src = state.image?.path?.let {
+                                            "/api/$v1/avatar$it"
+                                        }
+                                            ?: "img/undraw_profile.svg"
+                                        height = 260.0
+                                        width = 260.0
+                                    }
+                                }
+                                form {
+                                    val input = state.fieldsMap[InputTypes.USER_NAME] ?: ""
+                                    div {
+                                        inputTextFormRequired(
+                                            InputTypes.USER_NAME,
+                                            input,
+                                            (input.isEmpty() || input.isValidName()) && state.conflictErrorMessage == null,
+                                            "",
+                                            "User name",
+                                        ) {
+                                            changeFields(InputTypes.USER_NAME, it)
+                                            setState {
+                                                conflictErrorMessage = null
+                                            }
+                                        }
+                                    }
+                                    button {
+                                        type = ButtonType.button
+                                        className = ClassName("btn btn-info mt-4 mr-3")
+                                        +"Registration"
+                                        onClick = {
+                                            saveUser()
+                                        }
+                                    }
+                                    state.conflictErrorMessage?.let {
+                                        div {
+                                            className = ClassName("invalid-feedback d-block")
+                                            +it
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private fun postImageUpload(element: HTMLInputElement) =
+            scope.launch {
+                setState {
+                    isUploading = true
```

Comment:
Why isUploading needed here? Do you show any animation or what?

Reply:
Removed it